packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform.dart

```diff
@@ -50,54 +45,58 @@ class InAppPurchaseAndroidPlatform extends InAppPurchasePlatform {
   }
 
-  static late StreamController<List<PurchaseDetails>>
-      _purchaseUpdatedController;
+  final StreamController<List<PurchaseDetails>> _purchaseUpdatedController =
+      StreamController<List<PurchaseDetails>>.broadcast();
 
   @override
-  Stream<List<PurchaseDetails>> get purchaseStream =>
+  late final Stream<List<PurchaseDetails>> purchaseStream =
       _purchaseUpdatedController.stream;
 
   /// The [BillingClient] that's abstracted by [GooglePlayConnection].
   ///
   /// This field should not be used out of test code.
   @visibleForTesting
-  late final BillingClient billingClient;
+  final BillingClientManager billingClientManager = BillingClientManager();
 
-  late Future<void> _readyFuture;
   static final Set<String> _productIdsToConsume = <String>{};
 
   @override
   Future<bool> isAvailable() async {
-    await _readyFuture;
-    return billingClient.isReady();
+    return billingClientManager
+        .runRaw((BillingClient client) => client.isReady());
   }
 
   @override
   Future<ProductDetailsResponse> queryProductDetails(
       Set<String> identifiers) async {
     List<SkuDetailsResponseWrapper> responses;
     PlatformException? exception;
+
+    Future<SkuDetailsResponseWrapper> querySkuDetails(SkuType type) {
+      return billingClientManager.run(
+        (BillingClient client) => client.querySkuDetails(
+          skuType: type,
+          skusList: identifiers.toList(),
+        ),
+      );
+    }
+
     try {
       responses = await Future.wait(<Future<SkuDetailsResponseWrapper>>[
-        billingClient.querySkuDetails(
-            skuType: SkuType.inapp, skusList: identifiers.toList()),
-        billingClient.querySkuDetails(
-            skuType: SkuType.subs, skusList: identifiers.toList())
+        querySkuDetails(SkuType.inapp),
+        querySkuDetails(SkuType.subs),
       ]);
     } on PlatformException catch (e) {
       exception = e;
-      responses = <SkuDetailsResponseWrapper>[
-        // ignore: invalid_use_of_visible_for_testing_member
-        SkuDetailsResponseWrapper(
-            billingResult: BillingResultWrapper(
-                responseCode: BillingResponse.error, debugMessage: e.code),
-            skuDetailsList: const <SkuDetailsWrapper>[]),
-        // ignore: invalid_use_of_visible_for_testing_member
-        SkuDetailsResponseWrapper(
-            billingResult: BillingResultWrapper(
-                responseCode: BillingResponse.error, debugMessage: e.code),
-            skuDetailsList: const <SkuDetailsWrapper>[])
-      ];
+      // ignore: invalid_use_of_visible_for_testing_member
+      final SkuDetailsResponseWrapper response = SkuDetailsResponseWrapper(
+        billingResult: BillingResultWrapper(
+          responseCode: BillingResponse.error,
+          debugMessage: e.code,
+        ),
+        skuDetailsList: const <SkuDetailsWrapper>[],
+      );
+      responses = <SkuDetailsResponseWrapper>[response, response];
```

Comment:
Nice, this is good cleanup.

nit: Could you add a short comment above why this is leaving two identical responses. It wasn't immediately clear that they correspond with each of the queries done above.